wazuh/wazuh
# Purge Redhat vulnerabilities when the database is updated

|Related issue|
|---|
| #7043 |

## Description

This PR is a backport of the fix added at https://github.com/wazuh/wazuh/pull/7050. It fixes the following:

- Purge the Redhat vulnerabilities from the CVE database when an updated feed from the Redhat repositories is pulled. It prevents the database to grow with duplicated entries every time the feed was updated.
- Prevent to insert duplicated CVEs at SQL level. 

## Tests




- Compilation without warnings in every supported platform
- [x] scan-build 

### Manual testing

**1**. Configure Vulnerability Detector to pull Redhat vulnerabilities:

```xml
<provider name="redhat">
    <enabled>yes</enabled>
    <os>5</os>
    <os>6</os>
    <os>7</os>
    <os>8</os>
    <update_interval>1h</update_interval>
</provider>
```

**2**. Run Vulnerability Detector and wait until the DB has been populated.
**3**. Stop the Wazuh manager.
**4**. Modify the feed's timestamp `sqlite3 /var/ossec/queue/vulnerabilities/cve.db "UPDATE metadata set TIMESTAMP='2010-10-31T20:46:48' WHERE TARGET like 'RHEL%';"` to force an update
**5**. Restart the manager.
**6**. Once the vulnerabilities are updated, check the number of Redhat vulnerabilities. `sqlite3 /var/ossec/queue/vulnerabilities/cve.db "select count(*),TARGET from vulnerabilities group by TARGET;"`
